### ISSUE TITLE
Create Database Handling Thread

### ISSUE THREAD

**Issue by haerdib** [Polkadex-Substrate/polkadexTEE-worker#213]
Because the database writing will be a blocking call (no thread spawning possibl ewithin enclave) a separate thread (spawned in worker) will be handling the db struct. 

For the thread communcation a [Sender / Receiver](https://doc.rust-lang.org/std/sync/mpsc/index.html) could be used.
However, we need to create the mpsc channel within the enclave and the thread outside of the enclave.

So we have the following options:
- When initialising the enclave, a Database handling thread will be spawned in worker and then with an ecall sent into the enclave via ecall. In the enclave it will create an mpsc channel and create an inmemory storage (atomic pointer) and clone the sender into there. RPC threads in the enclave can then get the sender from this atomic pointer and send the data to be stored via channel.
- The RPC thread will then do an ocall to the worker and write the data to be stored into a buffer or directly to the data base
- Encoding / Decoding should be happening on worker side (RPC thread should be a fast as possible, also we need to make sure that single data base handling thread can handle a lot of parallel RPC threads)
- Maybe on worker side new thread(s) should be hanlding db storage (from buffer to db) - will need some further evaluation what is possible / necessary  / performance increasing

**Comment by arrudagates** [comment]
@haerdib Let me see if i understand the logic:
```
1 - create channel in enclave
2 - pass the receiver in an ocall to the worker
3 - in the worker receive the ocall and spawn a thread with the receiver
4 - when the enclave needs to store data to the DB it will use the sender
5 - when the thread in the worker receiver data from the enclave it will write to the DB
```

**Comment by haerdib** [comment]
Hm, no I don't think this will work. We can't pass an receiver per ocall to the worker. So the channel created within the enclave needs to stay there.
I've quickly noted down how I'm imaging things. Please take  a look at : https://raw.githubusercontent.com/haerdib/substraTEE_diagramms/main/database_thread.svg
(ofc, only very simplistic. On the rpc side I've omitted quite a lot of things, but it should be enough to demonstrate my thoughts)

Feel free do adapt to your needs. There might be a few things that have not yet been thought through. The only important thing : the rpc thread must get be blocked (resp. as little as possible) by the storage update.

**Comment by arrudagates** [comment]
Closed by #228
